drivers/ethernet/phy/phy_tja1103.c

```diff
@@ -153,140 +151,118 @@ static int phy_tja1103_id(const struct device *dev, uint32_t *phy_id)
 	return 0;
 }
 
-static int update_link_state(const struct device *dev)
+static int phy_tja1103_get_link_state(const struct device *dev, struct phy_link_state *state)
 {
 	struct phy_tja1103_data *const data = dev->data;
-	bool link_up;
 	uint16_t val;
+	int rc = 0;
+
+	k_sem_take(&data->sem, K_FOREVER);
```

Comment:
this semaphore needs to be around when the state in the data struct is changed. in phy_work_handler the state of the data struct is changed after this function.

Also it might be better for the function, that implements the `get_link` function of the phy api, to just just copy over the state from the data struct, instead of fetching it from the phy directly via mdio.